src/commands/deploy/metadata/quick.ts

```diff
@@ -92,15 +88,15 @@ export default class DeployMetadataQuick extends SfCommand<DeployResultJson> {
       return asyncFormatter.getJson();
     }
 
-    const result = await this.poll(jobId, flags.wait, componentSet);
+    const result = await poll(org, jobId, flags.wait, componentSet);
 
     const formatter = new DeployResultFormatter(result, flags);
 
-    if (!this.jsonEnabled()) {
-      formatter.display();
-    }
+    if (!this.jsonEnabled()) formatter.display();
 
-    await DeployCache.unset(jobId);
+    if (result.response.status === RequestStatus.Succeeded) {
```

Comment:
also...what about other terminal statuses (failure, partial success, cancelled, etc)?  Maybe we only want to keep incompletes in the cache/